packages/ui/src/components/Visualization/ContextToolbar/Flows/FlowsList.tsx

```diff
@@ -0,0 +1,117 @@
+import './FlowsList.css';
+
+import { Button, Icon } from '@patternfly/react-core';
+import { EyeIcon, EyeSlashIcon, TrashIcon } from '@patternfly/react-icons';
+
+import { FunctionComponent, useCallback, useContext, useRef } from 'react';
+import { Table, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+import { FlowsListEmptyState } from './FlowsListEmptyState';
+import { InlineEdit } from '../../../InlineEdit';
+
+import { BaseVisualCamelEntity } from '../../../../models/visualization/base-visual-entity';
+import { EntitiesContext } from '../../../../providers/entities.provider';
+
+interface IFlowsList {
+  onClose?: () => void;
+}
+
+export const FlowsList: FunctionComponent<IFlowsList> = (props) => {
+  const { visualEntities, visibleFlows, visualFlowsApi } = useContext(EntitiesContext)!;
+
+  const { isListEmpty, flows, setFlowName, deleteFlow } = {
+    isListEmpty: visualEntities.length === 0,
+    flows: visualEntities,
+    setFlowName: (id: string, name: string) => {
+      console.log('setting the name ', id, name);
+    },
+    deleteFlow: (id: string) => {
+      console.log('delete flow ', id);
+    },
+  };
+
+  const columnNames = useRef({
+    id: 'Route Id',
+    isVisible: 'Visibility',
+    delete: 'Delete',
+  });
+
+  const onSelectFlow = useCallback(
+    (flowId: string): void => {
+      visualFlowsApi.hideAllFlows();
+      visualFlowsApi.toggleFlowVisible(flowId);
+      props.onClose?.();
+    },
+    [props, visualFlowsApi],
+  );
+
+  return isListEmpty ? (
+    <FlowsListEmptyState data-testid="flows-list-empty-state" />
+  ) : (
+    <Table className="flows-list-table" variant="compact" data-testid="flows-list-table">
+      <Thead>
+        <Tr>
+          <Th>{columnNames.current.id}</Th>
+          <Th>{columnNames.current.isVisible}</Th>
+          <Th>{columnNames.current.delete}</Th>
+        </Tr>
+      </Thead>
+      <Tbody>
+        {flows.map((flow: BaseVisualCamelEntity) => (
+          <Tr key={flow.id} data-testid={`flows-list-row-${flow.id}`}>
+            <Td dataLabel={columnNames.current.id}>
+              <InlineEdit
+                data-testid={`goto-btn-${flow.id}`}
+                value={flow.id}
+                //  validator={ValidationService.validateUniqueName}
+                onClick={() => {
+                  onSelectFlow(flow.id);
+                }}
+                onChange={(name) => {
+                  setFlowName(flow.id, name);
+                }}
+              />
+              //TODO add description
+              <p>{''}</p>
```

Comment:
Maybe this is not necessary, as it shows on the UI
![image](https://github.com/KaotoIO/kaoto-next/assets/16512618/807b753c-cdad-42b9-ad6c-14ee7afa3513)



```suggestion
```